src/UCTNode.cpp

```diff
@@ -314,37 +315,52 @@ UCTNode* UCTNode::uct_select_child(int color, bool is_root) {
 
     const auto numerator = std::sqrt(double(parentvisits) *
             std::log(cfg_logpuct * double(parentvisits) + cfg_logconst));
-    const auto fpu_reduction = (is_root ? cfg_fpu_root_reduction : cfg_fpu_reduction) * std::sqrt(total_visited_policy);
-    // Estimated eval for unknown nodes = original parent NN eval - reduction
-    const auto fpu_eval = get_net_eval(color) - fpu_reduction;
 
     auto best = static_cast<UCTNodePointer*>(nullptr);
     auto best_value = std::numeric_limits<double>::lowest();
 
+    // Get winrates for visited children.
+    // This goes from best to worst policy value.
+    // An unvisited child gets the lowest winrate seen so far.
+    double winrates[m_children.size()];
+    int idx = -1;
+    double smallest_winrate = get_net_eval(color);
     for (auto& child : m_children) {
-        if (!child.active()) {
-            continue;
+        idx++;
+        if (child.get_visits() > 0) {
+            winrates[idx] = child.get_eval(color);
+            if (winrates[idx] < smallest_winrate) {
```

Comment:
`smallest_winrate = std::min(smalleste_winrate, winrates[idx]);` would be more straightforward (and maybe faster)